Prevent strace's "readlink" and "readlinkat" tests from failing due to the
additional system call made by glibc with the patch "glibc-dl-cache.patch"
applied (introduced in commit 52564e9).

These changes cause strace to report during these tests only system calls on
files contained in the test directory, effectively filtering out the
additional readlink/readlinkat call on "/proc/self/exe" and allowing the tests
to complete as normal.

diff --git a/tests/gen_tests.in b/tests/gen_tests.in
index 8b4e2e9..cc3ca63 100644
--- a/tests/gen_tests.in
+++ b/tests/gen_tests.in
@@ -623,8 +623,8 @@ quotactl-xfs-v	-v -e trace=quotactl
 read-write	-a15 -eread=0,5 -ewrite=1,4 -e trace=read,write -P read-write-tmpfile -P /dev/zero -P /dev/null
 readahead	-a1
 readdir	-a16
-readlink	-xx
-readlinkat	-xx
+readlink	-xx --trace-path=test.readlink.link
+readlinkat	-xx --trace-path=test.readlinkat.link
 reboot		-s 256
 recv-MSG_TRUNC	-a26 -e trace=recv
 recvfrom	-a35
diff --git a/tests/readlink.gen.test b/tests/readlink.gen.test
index 4263234..418691b 100755
--- a/tests/readlink.gen.test
+++ b/tests/readlink.gen.test
@@ -1,4 +1,4 @@
 #!/bin/sh -efu
-# Generated by ./tests/gen_tests.sh from ./tests/gen_tests.in (readlink -xx ); do not edit.
+# Generated by ./tests/gen_tests.sh from ./tests/gen_tests.in (readlink -xx --trace-path=test.readlink.link); do not edit.
 . "${srcdir=.}/init.sh"
-run_strace_match_diff -xx 
+run_strace_match_diff -xx --trace-path=test.readlink.link
diff --git a/tests/readlinkat.gen.test b/tests/readlinkat.gen.test
index d7de993..a48d590 100755
--- a/tests/readlinkat.gen.test
+++ b/tests/readlinkat.gen.test
@@ -1,4 +1,4 @@
 #!/bin/sh -efu
-# Generated by ./tests/gen_tests.sh from ./tests/gen_tests.in (readlinkat -xx ); do not edit.
+# Generated by ./tests/gen_tests.sh from ./tests/gen_tests.in (readlinkat -xx --trace-path=test.readlinkat.link); do not edit.
 . "${srcdir=.}/init.sh"
-run_strace_match_diff -xx 
+run_strace_match_diff -xx --trace-path=test.readlinkat.link
